### ISSUE TITLE
Execution failed for task ':mobile:transformClassesWithMamifyTransformForDebug'.  class androidx.work.impl.background.systemalarm.SystemAlarmService overrides final method com.microsoft.intune.mam.client.app.MAMService.onStartCommand

### ISSUE THREAD

**Issue by fredericojssilva** [msintuneappsdk/ms-intune-app-sdk-android#68]
**Describe the bug:**
After update to gradle 4.2.X start getting: 
```
Execution failed for task ':mobile:transformClassesWithMamifyTransformForDebug'.
 class androidx.work.impl.background.systemalarm.SystemAlarmService overrides final method com.microsoft.intune.mam.client.app.MAMService.onStartCommand(Landroid/content/Intent;II)I

```

**To Reproduce**

1. Update to gradle 4.2.1
2. Compile

**Expected behavior:**
It works

**Intune App SDK for Android:**
- What version of the Intune SDK are you using? Are you using the latest version? **SDK 7.2.2**
- What platform is your app based in (Java, Xamarin based, Cordova, etc)? **Java/Kotlin**
- For errors during build, does the app build without Intune SDK integration? **Yes**
- Do you see a trend with it only being reproduced on a specific device? *no device related*

**Comment by staticvoid7** [comment]
I am getting the same error building the project.

    FAILURE: Build failed with an exception.
    
    * What went wrong:
    Execution failed for task ':app:transformClassesWithMamifyTransformForDebug'.
    > class androidx.work.impl.background.systemalarm.SystemAlarmService overrides final method com.microsoft.intune.mam.client.app.MAMService.onStartCommand(Landroid/content/Intent;II)I

Do you need more information to fix this?

**Comment by yvan-botella** [comment]
Hi, we got exactly the same issue here too
Any news about it ?

**Comment by fredericojssilva** [comment]
`includeExternalLibraries = ["androidx.*"]`

Remove androidx from includeExternalLibraries seems to workaround the issue

**Comment by staticvoid7** [comment]
intunemam {
        includeExternalLibraries = ['androidx.*']
    }

We already have that configuration in our build.gradle. It is the default. Nonetheless we are getting the error.

**Comment by fredericojssilva** [comment]
@staticvoid7 I meant remove that configuration. Anyway, still don't know what it will affect, but going with this until its fixed on the sdk.

**Comment by staticvoid7** [comment]
At the moment our workaround is to use the old gradle 4.1.2.

**Comment by MathieuCalba** [comment]
Hello @fredericojssilva, removing `androidx.*` from `includeExternalLibraries` is not an option as this means Intune won’t intercept these classes.

**Comment by MathieuCalba** [comment]
It seems that `includeExternalLibraries` with the new version 7.6.0 of Intune plugin& SDK:
>The build plugin now automatically includes all external libraries when used with Android Gradle Plugin 4.2 and higher. These versions no longer expose the library names to the Transform API which our plugin is built on. The includeExternalLibraries configuration option will be removed in future MAM SDK 8.0.
https://github.com/msintuneappsdk/ms-intune-app-sdk-android/releases/tag/7.6.0

**Comment by bblia** [comment]
We are having the same problem on 7.6.0 with 4.2.x gradle versions. It appears that using those versions of gradle conflicts with the MAMification process so activities are not wrapped properly? We noticed that things like PIN code and copy/paste protection had just silently stopped working which is what led us down this path. Nothing worked except for downgrading to gradle 4.1.2. Hope this can get fixed in an upcoming release...

**Comment by meghandaly** [comment]
Underlying issues was the same as captured in #69 and has been resolved in the latest release, [version 7.6.0](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/releases/tag/7.6.0).
